src/reactivity/reactive.ts

```diff
@@ -118,11 +118,30 @@ export function observe<T>(obj: T): T {
   // in SSR, there is no __ob__. Mock for reactivity check
   if (!hasOwn(observed, '__ob__')) {
     def(observed, '__ob__', mockObserver(observed))
+    walk(observed)
   }
 
   return observed
 }
 
+/**
+ * Recursive obj, add __ob__
+ */
+function walk(obj: any) {
```

Comment:
```suggestion
function walk(obj: any, seen = new WeakMap<any, boolean>()) {
```

Would need a map to avoid circular reference.